scripts/extend-publish-config.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const { writeFileSync } = require('fs');
+const { resolve } = require('path');
+
+// This script applies the contents of any `extendedPublishConfig` blocks
+// to the root package.json of affected packages in PACKAGES_TO_EXTEND.
+//
+// It also removes `scripts` on the assumption none are needed when installing npm libraries.
+
+const PACKAGES_TO_EXTEND = ['workspace-resolver'];
```

Comment:
Technically this is directory names. I figured that having a hardcoded static list of packages to apply this to, is fine for the scope of this project.